Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp

```diff
@@ -411,6 +412,50 @@ JSValue JSInjectedScriptHost::getInternalProperties(JSGlobalObject* globalObject
         return array;
     }
 
+    if (JSFinalizationRegistry* finalizationRegistry = jsDynamicCast<JSFinalizationRegistry*>(value)) {
+        Locker locker { finalizationRegistry->cellLock() };
```

Comment:
You cannot hold a lock while doing JS operations like putDirect etc. it can cause deadlock.